### ISSUE TITLE
SVHN Baseline Accuracy 95.2%?

### ISSUE THREAD

**Issue by Mid-Push** [bruno-31/gan-manifold-reg#1]
Hi, Thanks for your work. 

I tested it on the SVHN experiments. But it seems like the model is able to achieve an accuracy of  95.2% only after 200 epoches, without manifold regularization.  However, your paper reports 95.37% with manifold regularization after 400 epoches.  So I'm wondering if the manifold regularization was not correctly implemented?

**Comment by bruno-31** [comment]
Hi ! Thank your interest.

You are raising an important point. You should not look at the performance of the classifier on the test set during training. Picking peak values on the test set may result in some overfitting on the test set. 

We have to use a validation set here, to know when to stop the training. We tried to run the experiments in a realistic setting (careful studies with a validation set is an open problem an many ssl paper do not use a validation set properly). We separated 10% of the training data and we designed our algorithms and chose our hyperparameters based on this validation set.  So the number of  training epochs is found on a validation set.  And we reported performance on the test set based on hyperparameters found on the validation set.

To answer your question, I observed that without the regularization, the neural net tends to reach high performance at some point of the training, but tends to overfit the training set and performances and is less stable.  So it stabilizes the training.
Hope that helps!

Bruno

**Comment by bruno-31** [comment]
Hi again,
After reading your comment I re-run some of the experiments.
Actually, we realized that the performance of the baseline for this setting is slightly underestimated.
So we'll fix the number for the next version.
Again thanks for flagging!
